src/modalities/config/config.py

```diff
@@ -132,9 +134,62 @@ class CheckpointingConfig(BaseModel):
     checkpointing_execution: PydanticCheckpointingExecutionIFType
 
 
+class AdamOptimizerConfig(BaseModel):
+    lr: float
+    wrapped_model: PydanticPytorchModuleType
+    betas: Tuple[float, float]
+    eps: float
+    weight_decay: float
+
+
 class AdamWOptimizerConfig(BaseModel):
     lr: float
     wrapped_model: PydanticPytorchModuleType
+    betas: Tuple[float, float]
+    eps: float
+    weight_decay: float
+
+
+class StepLRSchedulerConfig(BaseModel):
+    optimizer: PydanticOptimizerIFType
+    step_size: int
+    gamma: float
+    last_epoch: int = -1
+    verbose: bool = False
+
+
+class OneCycleLRSchedulerConfig(BaseModel):
+    optimizer: PydanticOptimizerIFType
+    max_lr: float | List
+    total_steps: int
+    epochs: Optional[int] = None
+    steps_per_epoch: Optional[int] = None
+    pct_start: float
+    anneal_strategy: str
+    cycle_momentum: bool = True
+    base_momentum: float | List = 0.85
+    max_momentum: float | List = 0.95
+    div_factor: float
+    final_div_factor: float
```

Comment:
we could also be more restrictive here:

`Annotated[int, Field(strict=True, gt=0)]`

same for the  other int and float types.

Reply:
Done in 8a4702d425cf8f8a4e3e351726c160d0317acf5c.